### ISSUE TITLE
Is there any way to prevent the orientation change?

### ISSUE THREAD

**Issue by kxgonzalez** [wcoder/Xamarin.Plugin.DeviceOrientation#13]
Hi,
I want to prevent the app to be rotated to portrait mode in certain scenarios, but I need it available so I can't disable it completely.
I know we have the OrientationChanged event but it fires after the orientation changes.

In other words; I want the app to be available in landscape and landscape flipped mode.
How can I achieve this?

I'm currently using Xamarin.Forms with PCL.

**Comment by rakeshostwal** [comment]
Hi @kxgonzalez , were you able to fix this?

**Comment by wcoder** [comment]
Hi @kxgonzalez 

Try to use `LockOrientation` method, for example:

```
CrossDeviceOrientation.Current.(DeviceOrientations.Landscape);
```

**Comment by wcoder** [comment]
Currently, there is no such functionality.

But you can setup platform manifests "supported device orientation" https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/layouts/device-orientation?tabs=vswin
for allow only landscape orientation

**Comment by rakeshostwal** [comment]
Yes, but that would set it on a complete application, right? I want to set only horizontal orientation on a specific screen.

**Comment by kxgonzalez** [comment]
I'm in the same situation.
I have a multipurpose app; one mode has views designed to be vertical. I have no problem with this.
The other mode is designed to be horizontal, so I want to be able to rotate it landscape and landscape inverted.
I tried with some logic but it fires after the rotation and animation happens.

**Comment by wcoder** [comment]
This situation is good for next version of the plugin.

I can propose the workaround for yours:

Manual override these methods on each platform:
### Android
`MainActivity.cs`
```
public override void OnConfigurationChanged(Android.Content.Res.Configuration newConfig)
```
### iOS
`AppDelegate.cs`
```
public override UIInterfaceOrientationMask GetSupportedInterfaceOrientations(UIApplication application, UIWindow forWindow)
```

In these methods, you can lock the device orientation for the required page.

**Comment by kxgonzalez** [comment]
Nice, thank you!
